src/main/java/com/hashedin/bitly/ShortCodeMappingService.java

```diff
@@ -0,0 +1,63 @@
+package com.hashedin.bitly;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ShortCodeMappingService {
+
+	private final Map<String, URI> mapping;
+	
+	public ShortCodeMappingService(@Value("classpath:mapping.properties") Resource mappingResource) throws IOException {
+		this.mapping = loadMapping(mappingResource);
+	}
+	
+	private static final Map<String, URI> loadMapping(Resource mappingResource) throws IOException {
+		Properties props = new Properties();
+		try(InputStream in = mappingResource.getInputStream()) {
+			props.load(mappingResource.getInputStream());			
+		}
+		
```

Comment:
We load the file using the inbuilt Properties class. Parsing files is painful, where possible use an existing format. It will handle many more edge cases.